app/src/main/java/woowacourse/payments/ui/component/NumberTextField.kt

```diff
@@ -0,0 +1,63 @@
+package woowacourse.payments.ui.component
+
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.text.KeyboardOptions
+import androidx.compose.material3.OutlinedTextField
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.TextRange
+import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.text.input.TextFieldValue
+import androidx.compose.ui.text.input.VisualTransformation
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import woowacourse.payments.R
+import woowacourse.payments.ui.theme.Gray100
+import woowacourse.payments.ui.theme.Gray200
+
+@Composable
+fun NumberTextField(
+    value: String,
+    onValueChange: (String) -> Unit,
+    label: Int,
+    placeholder: Int,
+    maxLength: Int,
+    modifier: Modifier = Modifier,
+    visualTransformation: VisualTransformation = VisualTransformation.None,
+) {
+    val textFieldValue =
+        remember(value) {
+            TextFieldValue(text = value, selection = TextRange(value.length))
+        }
+
+    OutlinedTextField(
+        value = textFieldValue,
+        onValueChange = { input ->
+            val digitsOnly = input.text.filter { it.isDigit() }.take(maxLength)
+            onValueChange(digitsOnly)
+        },
```

Comment:
```suggestion
    OutlinedTextField(
        value = value,
        onValueChange = { input ->
            val digitsOnly = input.filter { it.isDigit() }.take(maxLength)
            onValueChange(digitsOnly)
        },
```


> 숫자 입력을 할 때 자동으로 구분자가 들어가게 하니 커서가 구분자의 길이만큼 밀리는 현상

visualTransformation을 잘 적용해 주셔서 말씀하신 내용이 해결될 것 같네요😁